corehq/apps/data_interfaces/migrations/0036_backfill_dedupe_match_values.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 3.2.23 on 2023-12-12 01:18
+
+from django.db import migrations, transaction
+from corehq.form_processor.models.cases import CommCareCase
+from corehq.apps.data_interfaces.models import CaseDuplicate, CaseDuplicateNew
+
+
+def forwards_func(apps, schema_editor):
+    db_alias = schema_editor.connection.alias
+    process_records(db_alias=db_alias)
+
+
+def reverse_func(apps, schema_editor):
+    # Reverse isn't particularly important here, because generally we'd want to revert both this
+    # and the prior commit, dropping the table entirely
+    pass
+
+
+def process_records(batch_size=10, db_alias=None):
+    query = CaseDuplicate.objects.using(db_alias).all().order_by('id')
+    for (start, end, total, batch) in batch_qs(query, 100):
+        with transaction.atomic():
+            inserts = []
+            for duplicate in batch:
+                case = CommCareCase.objects.get_case(duplicate.case_id)
+                new_record = CaseDuplicateNew.create(case, duplicate.action, save=False)
+                inserts.append(new_record)
+            CaseDuplicateNew.objects.bulk_create(inserts, ignore_conflicts=True)
+
+
+# Copied from: https://djangosnippets.org/snippets/1170/
+def batch_qs(qs, batch_size=1000):
+    """
+    Returns a (start, end, total, queryset) tuple for each batch in the given
+    queryset.
+
+    Usage:
+        # Make sure to order your querset
+        article_qs = Article.objects.order_by('id')
+        for start, end, total, qs in batch_qs(article_qs):
+            print "Now processing %s - %s of %s" % (start + 1, end, total)
+            for article in qs:
+                print article.body
+    """
+    total = qs.count()
+    for start in range(0, total, batch_size):
+        end = min(start + batch_size, total)
+        yield (start, end, total, qs[start:end])
```

Comment:
Looks like this batching technique uses `LIMIT`/`OFFSET`, which is not very efficient for large data sets (which is, I believe, the entire point of batching here). We have a [`paginate_query`](https://github.com/dimagi/commcare-hq/blob/930687215a28ef775b01bd918c0067e335fad665/corehq/sql_db/util.py#L71) function that does batching efficiently, assuming there is an index on `pk`.

Reply:
Fixed in 7385e4a1f584334c5e61bab39e40d80d9d21732e